yocto/fetch-sources-for-recipes.sh

```diff
@@ -18,23 +18,28 @@
 # For further information see LICENSE
 #
 
+#
+# Fetches sources for a specific image with bitbake, retrying up to 100 times.
+# This is so that it will be possible to build even if one loses internet
+# connection after this step. We retry in case of temporary hickups.
+#
+
 YOCTO_DIR="$1"
 IMAGES="$2"
 COUNTER=100
 
 # Set up bitbake environment
 echo "Setting up bitbake environment."
-cd "$YOCTO_DIR/sources/poky/"
-source oe-init-build-env ../../build
+source "$YOCTO_DIR/sources/poky/oe-init-build-env" "$YOCTO_DIR/build"
 
 # Fetch all sources
 echo "Fetching all the sources. Will try $COUNTER times in case of a bad connection."
 I=0
-while [  $I -lt $COUNTER ]; do
+while [ $I -lt $COUNTER ]; do
     echo "Try number: $I"
     let I+=1
     sleep 1
-    time bitbake -c fetchall $IMAGES
+    time bitbake -k -c fetchall $IMAGES
```

Comment:
So the `-k` will make it fail later? Isn't it good to know early that one source is not available and the image will not be able to be build as early as possible? Would would be the purpouse of downloading the rest for almost one hour if you won't be able to build anyway?

Reply:
In this script, even if fetchall fails, we retry 100 times, so failing late makes sense from the perspective of getting as many sources as possible per retry, plus the delay might make some hickup-y server get well between tries if we're lucky, but I see your point as well, and there is merit to both views. Would you like me to change it back?

Reply:
Good point with the hick up, I don't really have a preference.